test/client_test.dart

```diff
@@ -25,4 +29,67 @@ void main() {
     expect(session.accessToken,
         'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJhdXRoZW50aWNhdGVkIiwiZXhwIjoxNjExODk1MzExLCJzdWIiOiI0Njg3YjkzNi02ZDE5LTRkNmUtOGIyYi1kYmU0N2I1ZjYzOWMiLCJlbWFpbCI6InRlc3Q5QGdtYWlsLmNvbSIsImFwcF9tZXRhZGF0YSI6eyJwcm92aWRlciI6ImVtYWlsIn0sInVzZXJfbWV0YWRhdGEiOm51bGwsInJvbGUiOiJhdXRoZW50aWNhdGVkIn0.GyIokEvKGp0M8PYU8IiIpvzeTAXspoCtR5aj-jCnWys');
   });
+
+  test('signUp()', () async {
+    final response = await client.signUp(email, password);
+    final data = response.data;
+    final error = response.error;
+    expect(error, isNull);
+    expect(data.accessToken is String, true);
+    expect(data.refreshToken is String, true);
+    expect(data.user.id is String, true);
+  });
+
+  test('signIn()', () async {
+    final response = await client.signIn(email: email, password: password);
+    final data = response.data;
+    final error = response.error;
+    expect(error, isNull);
+    expect(data.accessToken is String, true);
+    expect(data.refreshToken is String, true);
+    expect(data.user.id is String, true);
+  });
+
+  test('Get user', () async {
+    final user = client.user();
+    expect(user.id is String, true);
+    expect(user.appMetadata['provider'], 'email');
+  });
+
+  test('Update user', () async {
+    final response =
+        await client.update(UserAttributes(data: {'hello': 'world'}));
+    final data = response.data;
+    final error = response.error;
+    expect(error, isNull);
```

Comment:
FYI, here, the error is `message:"type 'String' is not a subtype of type 'Map<String, dynamic>' in type cast"`.

Reply:
fixed https://github.com/supabase/gotrue-dart/pull/3/commits/0a71c4c6ef70c0792e37e19988f9d135d76ccaec